### DISCUSSION TITLE
What is wrong with this command?

### DISCUSSION THREAD

**Discussion by SentientBeingOnAQuest** [ImageMagick/ImageMagick#4584] [Help] [answered]
For this command: 

`for %F in (*.jpg) do magick %F +write mpr:img -bordercolor Black -border 2x2 +write mpr:img -blur 40x10 -resize x640 -gravity center  -crop 600x600+0+0 +repage mpr:img -gravity center -compose over -composite result.jpg`

I am getting the following error messages:

magick: unable to open image 'maxresdefault': No such file or directory @ error/blob.c/OpenBlob/3527.
magick: no decode delegate for this image format `' @ error/constitute.c/ReadImage/572.

How can I fix this command? (Win10 IM7)

**Comment by snibgo** [comment]
I can't see any problem with your command. Perhaps there is a strange file on your computer. Try the following:
```
( for %F in (*.jpg) do echo %F ) >x.txt
```
That creates a file x.txt with a list of filenames that match *.jpg. Is there a file named maxresdefault?

EDIT: Perhaps you have spaces in your filenames. If so, then adjust your command:
```
for "tokens=*" %F in (*.jpg) do magick "%F" +write mpr:img ...
```

**Reply by SentientBeingOnAQuest** [reply]
OMG I've spent days trying to figure out why only two of the files would actually register, it never occurred to me that these were the only two files which didn't have spaces in the filename!  I changed all the filenames with bulkrename utility, and now all files are being processed.

The only problem now is that it only creates 1 output file which is being overwritten by every next file. How can I get it to overwrite the input files or create a copy for every file and maintain the file name?

Also for `for "tokens=*" %F` I'm getting the following warning: "tokens=*" was unexpected at this time. I've tried taking out the " but that doesn't work either. Any suggestions? 

I'm going to mark your reply as answer anyway, because you solved the biggest part of the puzzle for me.

**Comment by snibgo** [comment, marked as answer]
Sorry, I was too lazy to check the syntax of "for". Remove "tokens=*", so the command starts:
```
for %F in (*.jpg) do magick "%F" +write mpr:img ...
```
The important bit is the quotes around %F in the magick command.

For the output filename, I suggest using a different directory. (EDIT: This directory must exist before you run the magick command.) Then you can re-run the command without re-processing the outputs you have just created.
```
... -composite "mynewdir\%F"
```
Or you could use a modified version of the filename, such as:
```
... -composite "result_%F"
```
Or combine these methods. The output filename needs to be quoted if %F contains spaces.

**Reply by SentientBeingOnAQuest** [reply]
Thank you so much! Seriously, I can't thank you enough. I've been struggling with this for about a week now.
This works fantastic! 

I've marked this reply as the answer, as it's more complete than your earlier comment.
